Automattic/pocket-casts-ios
# [Paywall Experiment] Add dynamic header based on source

| 📘 Part of: #1986
|:---:|

This PR updates the default paywall header based on the source type. The header changes if the paywall is presented by tapping a specific source:

https://github.com/user-attachments/assets/0cfcc0b7-65a9-4c2a-ac13-f2e9b8fc6405

The sources are:
1. folders
2. upNextShuffle
3. themes
4. watch
5. icons
6. files

Any other sources applies the default header.

This PR also limits the AB experiment if the source is profile or the onboarding

## To test

- CI must be 🟢 
- Make sure the FF `upgradeExperiment` is on
- Login with a non plus user
- Test the sources listed above (Check the video to see what sources trigger the different headlines)
- Assign yourself to an experiment checking our P2 for instruction
- Go to profile, scroll down and tap on `Learn more about.....`
- You should see the Paywall with the variation assigned to you

## Checklist
